### ISSUE TITLE
Fails on android 12 when use --stay-awake

### ISSUE THREAD

**Issue by binlly** [Genymobile/scrcpy#2671]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: macOS BigSur 11.6
 - scrcpy version: 1.19
 - installation method: brew
 - device model: Xiaomi 11 Ultra
 - Android version: 12

**Describe the bug**

run: scrcpy --stay-awake

```
2021-09-28 17:49:19.443 scrcpy[48746:3478246] INFO: scrcpy 1.19 <https://github.com/Genymobile/scrcpy>
/usr/local/share/scrcpy/scrcpy-server: 1 file pushed, 0 skipped. 30.8 MB/s (37442 bytes in 0.001s)
[server] INFO: Device: Xiaomi (Android 12)
[server] ERROR: Could not invoke method
java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.genymobile.scrcpy.wrappers.ContentProvider.call(ContentProvider.java:108)
	at com.genymobile.scrcpy.wrappers.ContentProvider.getValue(ContentProvider.java:149)
	at com.genymobile.scrcpy.wrappers.ContentProvider.getAndPutValue(ContentProvider.java:165)
	at com.genymobile.scrcpy.Server.scrcpy(Server.java:39)
	at com.genymobile.scrcpy.Server.main(Server.java:255)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:363)
Caused by: java.lang.SecurityException: Calling uid: 2000 doesn't match source uid: 0
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
	at android.os.Parcel.createException(Parcel.java:2409)
	at android.os.Parcel.readException(Parcel.java:2392)
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:190)
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:142)
	at android.content.ContentProviderProxy.call(ContentProviderNative.java:732)
	... 8 more
Caused by: android.os.RemoteException: Remote stack trace:
	at android.content.AttributionSource.enforceCallingUid(AttributionSource.java:239)
	at android.content.AttributionSource.<init>(AttributionSource.java:158)
	at android.content.AttributionSource$1.createFromParcel(AttributionSource.java:454)
	at android.content.AttributionSource$1.createFromParcel(AttributionSource.java:446)
	at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:289)

[server] ERROR: Could not invoke method
java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.genymobile.scrcpy.wrappers.ContentProvider.call(ContentProvider.java:108)
	at com.genymobile.scrcpy.wrappers.ContentProvider.putValue(ContentProvider.java:161)
	at com.genymobile.scrcpy.wrappers.ContentProvider.getAndPutValue(ContentProvider.java:167)
	at com.genymobile.scrcpy.Server.scrcpy(Server.java:39)
	at com.genymobile.scrcpy.Server.main(Server.java:255)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:363)
Caused by: java.lang.SecurityException: Calling uid: 2000 doesn't match source uid: 0
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
	at android.os.Parcel.createException(Parcel.java:2409)
	at android.os.Parcel.readException(Parcel.java:2392)
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:190)
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:142)
	at android.content.ContentProviderProxy.call(ContentProviderNative.java:732)
	... 8 more
```

**Comment by rom1v** [comment]
Is your device rooted?

For some reasons, some features do not work if scrcpy is run as root.

**Comment by binlly** [comment]
It seems to pass when I set USER_ID:2000
But the followed message printed:

```
2021-09-28 18:22:21.975 scrcpy[50024:3495623] INFO: scrcpy 1.19 <https://github.com/Genymobile/scrcpy>
/usr/local/share/scrcpy/scrcpy-server: 1 file pushed, 0 skipped. 33.2 MB/s (37490 bytes in 0.001s)
[server] INFO: Device: Xiaomi (Android 12)
[server] ERROR: Exception on thread Thread[main,5,main]
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.genymobile.scrcpy.wrappers.ContentProvider.getAndPutValue(java.lang.String, java.lang.String, java.lang.String)' on a null object reference
	at com.genymobile.scrcpy.Server.scrcpy(Server.java:39)
	at com.genymobile.scrcpy.Server.main(Server.java:255)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:363)
```

**Comment by rom1v** [comment]
> It seems to pass when I set USER_ID:2000

Where do you set this?

**Comment by binlly** [comment]
> Is your device rooted?
> 
> For some reasons, some features do not work if scrcpy is run as root on the device.

Not root

**Comment by binlly** [comment]
> > It seems to pass when I set USER_ID:2000
> 
> Where do you set this?

I try modified the source code and ran it locally

**Comment by binlly** [comment]
I think I know what's going on here. Maybe it really has something to do with android system permissions.

**Comment by rom1v** [comment]
> I try modified the source code and ran it locally

What change did you do exactly?

**Comment by binlly** [comment]
I probably realized the problem (my system MIUI is dev channel), but some attempt ended in failure. Maybe xiaomi did something to cause it.
It might be a good idea to temporarily close this issue. Thank you.

**Comment by rom1v** [comment]
Please test #2802.

**Comment by Tyf0x** [comment]
This is working, thanks @rom1v  🎉

```
INFO: scrcpy 1.20 <https://github.com/Genymobile/scrcpy>
* daemon not running; starting now at tcp:5037
* daemon started successfully
D:\softs\scrcpy\scrcpy-server: 1 file pushed, 0 skipped. 14.1 MB/s (38647 bytes in 0.003s)
[server] INFO: Device: Google Pixel 3 XL (Android 12)
[server] INFO: Device screen turned off
INFO: Renderer: opengl
INFO: OpenGL version: 4.6.0 NVIDIA 496.76
INFO: Trilinear filtering enabled
INFO: Initial texture: 800x392
```

**Comment by rom1v** [comment]
:+1: Fixed by #2802.
